lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -354,25 +355,24 @@ internal void RecordRunningTaskDuringSystemFailure(IRunningTask runningTask, str
         }
 
         /// <summary>
-        /// Gets error type based on the information in IFailedTask 
-        /// Currently we use the Message in IFailedTask to distinguish different types of errors
+        /// Gets error type based on the exception type in IFailedTask 
         /// </summary>
         /// <param name="failedTask"></param>
         /// <returns></returns>
-        private TaskStateEvent GetTaskErrorEvent(IFailedTask failedTask)
+        private TaskStateEvent GetTaskErrorEventByExceptionType(IFailedTask failedTask)
         {
-            switch (failedTask.Message)
+            if (failedTask.AsError() is IMRUTaskAppException)
```

Comment:
Would it be worthwhile to cache the result of `failedTask.AsError()` in a local variable?